moralization/plot.py

```diff
@@ -108,165 +110,352 @@ def _generate_corr_df(occurence_df: pd.DataFrame, _filter=None) -> pd.DataFrame:
 
-    def show(self):
-        """Display the interactive plot"""
-        IPython.display.display(self._display_container)
+        self.app.callback(
+            Output("graph_output", "figure"),
+            Input("dropdown_analyzer_key", "value"),
+            Input("dropdown_analyzer_span_cat", "value"),
+        )(self.update_graph)
 
-    def visualize_analyzer_result(self, span_label="sc", analysis_type="frequency"):
+    def change_analyzer_key(self, key_input):
         """
+        Changes the analyzer key in response to a user selection.
 
         Args:
-            data_manager_all (dict): The returned analyzer results for all 4 types as a dict.
-            span_label (str, optional): The span label which should be shown. Defaults to "sc".
-            analysis_type (str, optional): Can be `frequency`, `length`,
-                `span_distinctiveness` or `boundary_distinctiveness`. Defaults to "frequency".
+            key_input: The user-selected analyzer key.
+
+        Returns:
+            A tuple containing the analyzer span categories and the first category in the list.
         """
+        analyzer_span_cat = sorted(list(self.analyzer_results_all[key_input].keys()))
+        return analyzer_span_cat, analyzer_span_cat[0]
 
-        if analysis_type not in list(self.analyzer_results_all.keys()):
-            raise KeyError(
-                f"key {analysis_type} not in analyzer_results_all, which only has"
-                + f"{list(self.analyzer_results_all.keys())}"
-            )
+    def update_graph(self, input_analyzer_key, input_analyzer_span_cat):
+        """
+        Updates the graph in response to a user selection.
+
+        Args:
+            input_analyzer_key: The user-selected analyzer key.
+            input_analyzer_span_cat: The user-selected analyzer span category.
 
-        if span_label not in self.analyzer_results_all[analysis_type].keys():
-            raise KeyError(f"{span_label} is not present in the given dataset.")
+        Returns:
+            A plotly figure.
+        """
+        analyzer_result = self.analyzer_results_all[input_analyzer_key][
+            input_analyzer_span_cat
+        ]
+        fig = px.scatter(
+            analyzer_result,
+            labels={"value": input_analyzer_key, "variable": "chosen categories"},
+        )
+        return fig
 
-        df = self.analyzer_results_all[analysis_type][span_label]
-        # we don't need this column.
-        df = df.drop("paragraph")
+    def run_app(self):
+        """
+        Displays the interactive plot.
+        """
+        if not is_interactive():
+            raise EnvironmentError(
+                "Dash GUI is only available in an Ipython environment like Jupyter notebooks."
+            )
 
-        plt.figure(figsize=(10, 6))
-        plt.title(f"span {analysis_type}")
-        plt.xticks(rotation=90)
-        sn.scatterplot(df)
-        return plt.show()
+        return self.app.run_server(
```

Comment:
Can we also pass the port number if needed, in case there are two instances of the interactive interface?